pkg/kubectl/cmd/util/cached_discovery.go

```diff
@@ -231,8 +242,35 @@ func (d *CachedDiscoveryClient) ServerVersion() (*version.Info, error) {
 	return d.delegate.ServerVersion()
 }
 
+type cachedOpenAPIObjectEncoder struct{}
+
+func (e *cachedOpenAPIObjectEncoder) Encode(obj interface{}) ([]byte, error) {
+	return proto.Marshal(obj.(proto.Message))
+}
+
 func (d *CachedDiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
-	return d.delegate.OpenAPISchema()
+	filename := filepath.Join(d.cacheDirectory, "openapischema.json.pb")
```

Comment:
what part of this is json?  json does not look correct here.

Reply:
Thanks, updated to just `openapischema.pb`